### PR TITLE
multifile reader

### PR DESCRIPTION
**Pull request by Karsten1987**
mainly work of @piraka9011 based on #187

### CODE REVIEW

**Comment by Karsten1987** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8657/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4499/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7063/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8575/)

File: rosbag2/src/rosbag2/readers/sequential_reader.cpp [hunk 1]
```diff
@@ -108,4 +110,56 @@ std::vector<TopicMetadata> SequentialReader::get_all_topics_and_types()
...
+
+std::string SequentialReader::get_current_file() const
+{
+  return *current_file_iterator_;
+}
+
+std::string SequentialReader::get_current_uri() const
+{
+  auto current_file = get_current_file();
+  auto current_uri = rcpputils::fs::remove_extension(current_file);
```
Comment (piraka9011) [inline rosbag2/src/rosbag2/readers/sequential_reader.cpp]: This will need to be updated if we have compression extensions (ex. *.db3.snappy)

Comment (Karsten1987) [reply rosbag2/src/rosbag2/readers/sequential_reader.cpp]: In fact this is current not being used. The current behavior of the sqlite plugin requires a full path to the database when opening for read-only.

**Comment by Karsten1987** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8658/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4500/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7068/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8578/)

**Pull request merged**